src/Aspire.Dashboard/Model/ResourceTypeDetails.cs

```diff
@@ -1,10 +1,37 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using Aspire.Dashboard.Otlp.Model;
 
 namespace Aspire.Dashboard.Model;
 
-public record ResourceTypeDetails(OtlpApplicationType? Type, string? InstanceId);
+public class ResourceTypeDetails
+{
+    public ResourceTypeDetails(OtlpApplicationType? type, string? instanceId)
+    {
+        // Double check that the replica's have the right type.
+        // TODO: This feels pretty hacky. Consider refactoring.
+        if (type == OtlpApplicationType.Replica && !typeof(ResourceTypeDetails).IsAssignableFrom(GetType()))
```

Comment:
Using a factory pattern would let you avoid having to runtime check the OtlpApplicationType all over the place.

Something like:
```C#
public static ResourceTypeDetails CreateResourceTypeDetails(OtlpApplicationType? type, ResourceViewModel resource)
{
    return type switch
    {
        OtlpApplicationType.Replica => new ReplicaTypeDetails(type, resource.Name, resource.DisplayName),
        _ => new ResourceTypeDetails(type, resource.Name),
    }
}
```

Reply:
I removed inheritance and switched to factory methods for everything. Simpler and safer.